[Document] Usage of Dart-Define variables in gradle scripts

While creating **integration tests** for our Flutter application, we're facing a blocking issue:

Following the [official flutter documentation for integration testing](https://docs.flutter.dev/testing/integration-tests), in the section to automate the tests on Firebase Test Lab, its stated to [build the app with gradle](https://docs.flutter.dev/testing/integration-tests).

```bash
pushd android
# flutter build generates files in android/ for building the app
flutter build apk
./gradlew app:assembleAndroidTest
./gradlew app:assembleDebug -Ptarget=integration_test/<name>_test.dart
popd
```

Unfortuantely, its not stated on how to include `--dart-define` variables into the script so that the created APK is build with the variable defined.

According to a [stackoverflow thread](https://stackoverflow.com/a/67751484/5498155), I've tried setting the variable in `gradle.properties`, with no effect.

As nothing in this matter is documented, I assume its not possible and therefor file this issue.
If it is possible, how can we pass the `dart-define` to the gradle build? Adjusting the documentation here might be helpful :)


**Expected results:** <!-- what did you want to see? -->
Integration testable APK with a `dart-define` variable modified in its source code.

**Actual results:** <!-- what did you see? -->
The APK only has default behavior, the variable can't be modified.


<details>
  <summary>Logs</summary>

<!-- Finally, paste the output of running `flutter doctor -v` here. -->

```
[✓] Flutter (Channel stable, 2.10.3, on Microsoft Windows [Version 10.0.22000.556], locale de-DE)
    • Flutter version 2.10.3 at X:\Flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 7e9793dee1 (2 weeks ago), 2022-03-02 11:23:12 -0600
    • Engine revision bd539267b4
    • Dart version 2.16.1
    • DevTools version 2.9.2

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at X:\SDK
    • Platform android-31, build-tools 31.0.0
    • Java binary at: X:\JetBrains\apps\AndroidStudio\ch-0\211.7628.21.2111.8193401\jre\bin\java
    • Java version OpenJDK Runtime Environment (build 11.0.11+9-b60-7590822)
    • All Android licenses accepted.
```

</details>


<!--
      Consider also attaching screenshots and/or videos to better
      illustrate the issue.

      You can upload them directly on GitHub.
      Beware that video file size is limited to 10MB.
-->

I'm having a problem with passing dart defines to the `app:connectedDebugAndroidTest` task, which is needed to run integration tests natively on Android.

```
./gradlew app:connectedDevDebugAndroidTest \
  -Ptarget=$(pwd)/../integration_test/hsv_test.dart \
  -Pdart-defines="Q1RTX0RJU0FCTEVfTUFQX1BSRUxPQURJTkc9dHJ1ZQo=" 
```

but it's spitting out really ugly error message:

<details>
<summary>error message</summary>

```
> Task :app:compileFlutterBuildDevDebug
-DCTS_DISABLE_MAP_PRELOADING=true%0A: Error: Error when reading '-DCTS_DISABLE_MAP_PRELOADING=true%0A': No such file or directory


> Task :app:compileFlutterBuildDevDebug FAILED

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* Where:
Script '/Users/bartek/fvm/versions/stable/packages/flutter_tools/gradle/flutter.gradle' line: 1159

* What went wrong:
Execution failed for task ':app:compileFlutterBuildDevDebug'.
> Process 'command '/Users/bartek/fvm/versions/stable/bin/flutter'' finished with non-zero exit value 1

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* What went wrong:
java.lang.StackOverflowError (no error message)

* Try:
> Run with --stacktrace option to get the stack trace.
> Run with --info or --debug option to get more log output.
> Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

Deprecated Gradle features were used in this build, making it incompatible with Gradle 8.0.

You can use '--warning-mode all' to show the individual deprecation warnings and determine if they come from your own scripts or plugins.

See https://docs.gradle.org/7.5.1/userguide/command_line_interface.html#sec:command_line_warnings

BUILD FAILED in 7s
1 actionable task: 1 executed
```
<details>

--- a/packages/integration_test/README.md
+++ b/packages/integration_test/README.md
@@ -222,8 +222,8 @@ physical):
 ```
 
 Note:
-If you want to use `--dart-define` with `gradlew` you have to `base64` encode
-all parameters, and pass them to gradle in a comma separated list:
+To use `--dart-define` with `gradlew` you must `base64` encode all parameters, 
+and pass them to gradle in a comma separated list:
 ```bash
 ./gradlew project:task -Pdart-defines="{base64(key=value)},[...]"
 ```

--- a/packages/integration_test/README.md
+++ b/packages/integration_test/README.md
@@ -221,6 +221,13 @@ physical):
 ./gradlew app:connectedAndroidTest -Ptarget=`pwd`/../integration_test/foo_test.dart
 ```
 
+Note:
+If you want to use `--dart-define` with `gradlew` you have to `base64` encode
+all parameters, and pass them to gradle in a comma separated list:
+```bash
+./gradlew project:task -Pdart-defines="{base64(key=value)},[...]"
+```
+
 ## Firebase Test Lab
 
 If this is your first time testing with Firebase Test Lab, you'll need to follow

--- a/packages/integration_test/README.md
+++ b/packages/integration_test/README.md
@@ -222,7 +222,7 @@ physical):
 ```
 
 Note:
-To use `--dart-define` with `gradlew` you must `base64` encode all parameters, 
+To use `--dart-define` with `gradlew` you must `base64` encode all parameters,
 and pass them to gradle in a comma separated list:
 ```bash
 ./gradlew project:task -Pdart-defines="{base64(key=value)},[...]"

